daffodil-runtime1/src/main/scala/org/apache/daffodil/debugger/InteractiveDebugger.scala

```diff
@@ -1595,7 +1595,22 @@ class InteractiveDebugger(runner: InteractiveDebuggerRunner, eCompilers: Express
         val desc = "display the current Daffodil " + name
         val longDesc = desc
         def act(args: Seq[String], state: ParseOrUnparseState, processor: Processor): DebugState.Type = {
-          debugPrintln("%s: %s".format(name, processor.toBriefXML(2))) // only 2 levels of output, please!
+          state match {
+            case pstate: PState => {
+              if (name == "parser") {
+                debugPrintln("%s: %s".format(name, processor.toBriefXML(2)))
+              } else {
+                debugPrintln("unparser: not available")
+              }
+            }
+            case ustate: UState => {
+              if (name == "unparser") {
+                debugPrintln("%s: %s".format(name, processor.toBriefXML(2)))
+              } else {
+                debugPrintln("parser: not available")
+              }
+            }
+          }
```

Comment:
Just a thought, I wonder if there is value in adding a new `info processor` command, which would print out either the parser or unparser depending on which is available (i.e. the current behavior of the InfoParser and InfoUnparser commands)?

And then the "not available" logic would move to the individual parser/unparse commands. E.g.

```scala
class InfoParser ... {
  def act(...) = {
    state match {
      case _: PState => debugPrintln("%s: %s".format(name, processor.toBriefXML(2)))
      case _ => debugPrintln("%s: not available".format(name))
    }
  }
}
```

InfoUnparser would be the same but would match on `UState` instead of `PState`.

We could also deprecate info parser/unparser and just have the info processor command if we wanted. I'm not sure there's much benefit from having both when they do basically the same thing.

If there's consensus that this seems reasonable, we can either add it to this PR, or just open a new feature ticket to implement some other day. I have no strong preference.

Reply:
Suggest we integrate this as is, but that next change should be another ticket, but might as well get it done right away while all fresh in mind. 

Why would I suggest this?... because the world seems full of interrupts of context to me right now.